EngineAPI/src/main/java/nullengine/event/world/block/cause/BlockChangeCause.java

```diff
@@ -1,7 +1,20 @@
 package nullengine.event.world.block.cause;
 
+import nullengine.entity.Entity;
 import nullengine.event.cause.Cause;
+import nullengine.item.Item;
+import nullengine.world.World;
 
 public interface BlockChangeCause extends Cause {
+    interface EntityTrigger extends BlockChangeCause {
+        Entity getEntity();
+    }
```

Comment:
These are all examples of how these things will be.

Please have some suggestions @Mouse0w0  @cvrunmin

Reply:
That's enough. I have nothing to say.